exist-core/src/main/java/org/exist/xquery/util/SerializerUtils.java

```diff
@@ -411,37 +413,44 @@ private static void setPropertyForMap(final Properties properties, final Paramet
+        final String value;
+
         switch (parameterConvention.getType()) {
             case Type.BOOLEAN:
-                properties.setProperty(
-                        parameterConvention.getLocalParameterName(),
-                        ((BooleanValue) parameterValue.itemAt(0)).getValue() ? "yes" : "no"
-                );
+                value = ((BooleanValue) parameterValue.itemAt(0)).getValue() ? "yes" : "no";
+                properties.setProperty(localParameterName, value);
                 break;
             case Type.STRING:
-                properties.setProperty(parameterConvention.getLocalParameterName(), ((StringValue) parameterValue.itemAt(0)).getStringValue());
+                value = ((StringValue) parameterValue.itemAt(0)).getStringValue();
+                properties.setProperty(localParameterName, value);
                 break;
             case Type.DECIMAL:
-                properties.setProperty(parameterConvention.getLocalParameterName(), ((DecimalValue) parameterValue.itemAt(0)).getStringValue());
+                value = ((DecimalValue) parameterValue.itemAt(0)).getStringValue();
+                properties.setProperty(localParameterName, value);
                 break;
             case Type.QNAME:
                 if (Cardinality._MANY.isSuperCardinalityOrEqualOf(parameterConvention.getCardinality())) {
                     final SequenceIterator iterator = parameterValue.iterate();
                     while (iterator.hasNext()) {
-                        final String existingValue = (String) properties.get(parameterConvention.getLocalParameterName());
+                        final String existingValue = (String) properties.get(localParameterName);
+                        final String nextValue = ((QNameValue) iterator.nextItem()).getQName().toURIQualifiedName();
+
                         if (existingValue == null || existingValue.isEmpty()) {
-                            properties.setProperty(parameterConvention.getLocalParameterName(), ((QNameValue) iterator.nextItem()).getQName().toURIQualifiedName());
+                            properties.setProperty(localParameterName, nextValue);
                         } else {
-                            properties.setProperty(parameterConvention.getLocalParameterName(), existingValue + " " + ((QNameValue) iterator.nextItem()).getQName().toURIQualifiedName());
+                            properties.setProperty(localParameterName, existingValue + " " + nextValue);
                         }
                     }
                 } else {
-                    properties.setProperty(parameterConvention.getLocalParameterName(), ((QNameValue) parameterValue.itemAt(0)).getQName().toURIQualifiedName());
+                    value = ((QNameValue) parameterValue.itemAt(0)).getQName().toURIQualifiedName();
+                    properties.setProperty(localParameterName, value);
                 }
                 break;
             case Type.MAP:
```

Comment:
missing default: ?

Reply:
There was none before, what would be an appropriate default here? I assume a `UsupportedOperationException` as in case of `Type.Map`.

Reply:
@dizzzz What do you think?

Reply:
Sorry I missed it;
So yes we should be cautious in not explicitly handling situation; parallel t the Map sounds good to me.

Reply:
Added the default to guard against unsupported types in ParameterConventions (these are defined in the class itself so it is rather theoretical, but regardless it will help future devs in not making this mistake.)